api.go

```diff
@@ -45,6 +68,30 @@ func parseAvailableCarResponse(res *http.Response) (carAvailabilitiesResp, error
 	return parsedBody, err
 }
 
+func GetAvailableFlexCars(ctx context.Context, query CarQuery) (flexCarAvailabilitiesResp, error) {
```

Comment:
Should do the same as `GetAvailableCars` and convert the communauto API format to a `Car[]`. The goal is that even if the communauto api changes, the users of this package don't have to change their implementation